src/pages/index.js

```diff
@@ -105,36 +138,75 @@ function handleAddNewCardButton() {
 }
 
 function handleProfileFormSubmit(inputValues) {
+  document.querySelector(settings.submitButtonSelector).textContent =
+    "Saving...";
+  api
+    .editUserInfo(inputValues)
+    .then((data) => {
+      // process the result
+      const name = data.name;
+      const subtitle = data.about;
+      return { name, subtitle };
+    })
+    .finally(() => {
+      document.querySelector(settings.submitButtonSelector).textContent =
+        "Save";
+    });
   userInfo.setUserInfo(inputValues);
   addProfilePopup.closeModal();
 }
 
+function handleCardDeleteFunctionInIndexComponent(cardId) {
+  api.deleteCard(cardId).then(() => {});
+}
+
 function handleAddNewCardFormSubmit(inputValues) {
-  //create a new card with input values from user
-  const card = createCard(inputValues);
-  //Attach new card to begining of container
-  section.prependItem(card);
-  //close popup after submit
-  newCardPopup.closeModal();
+  //set button to Saving.. while api call is made
+  document.querySelector("#create-button").textContent = "Saving...";
+  //create a new card with input values from server
+  api
+    .addNewCard(inputValues)
+    .then((data) => {
+      const card = createCard(data);
+      //Attach new card to begining of container
+      section.prependItem(card);
+      //close popup after submit
+      newCardPopup.closeModal();
+    })
+    .finally(() => {
+      document.querySelector("#create-button").textContent = "Create";
+    });
 }
 
 function onCardClick(card) {
   cardImagePopup.openModal(card);
+  caption.textContent = card.name;
```

Comment:
`  caption.textContent = card.name;
`

This action should be inside method `openModal` in class `PopupWithImage`.